lib/msf/core/payload/adapter/fetch/server/https.rb

```diff
@@ -0,0 +1,144 @@
+module Msf::Payload::Adapter::Fetch::Server::Https
+
+  # This mixin supports both HTTP and HTTPS fetch handlers.  If you only want
+  # HTTP, use the HTTP mixin that imports this, but removes the HTTPS options
+  def initialize(*args)
+    super
+    register_options(
+      [
+        Msf::OptBool.new('FETCH_CHECK_CERT', [true,"Check SSL certificate", false])
+
+      ]
+    )
+    register_advanced_options(
+      [
+        Msf::OptString.new('FetchHttpServerName', [true, 'Http Server Name', 'Apache']),
+        Msf::OptPath.new('FetchSSLCert', [ false, 'Path to a custom SSL certificate (default is randomly generated)', '']),
+        Msf::OptBool.new('FetchSSLCompression', [ false, 'Enable SSL/TLS-level compression', false ]),
+        Msf::OptString.new('FetchSSLCipher', [ false, 'String for SSL cipher spec - "DHE-RSA-AES256-SHA" or "ADH"']),
+        Msf::OptEnum.new('FetchSSLVersion',
+                         'Specify the version of SSL/TLS to be used (Auto, TLS and SSL23 are auto-negotiate)',
+                         enums: Rex::Socket::SslTcp.supported_ssl_methods)
+      ]
+    )
+  end
+
+  def add_resource(fetch_service, uri)
+    if fetch_service.resources.include?(uri)
+      fail_with(Msf::Exploit::Failure::BadConfig, "Resource collision detected.  Set FETCH_URI to a different value to continue.")
+    end
+    fetch_service.add_resource(uri,
+                               'Proc' => proc do |cli, req|
+                                 on_request_uri(cli, req, srvexe)
+                               end,
+                               'VirtualDirectory' => true)
+  rescue  ::Exception => e
+    fetch_service.deref
+    cleanup_handler
+    fail_with(Msf::Exploit::Failure::Unknown, "Failed to add resource\n #{e}")
+  end
+
+  def cleanup_http_fetch_service(fetch_service)
+    unless fetch_service.nil?
+      fetch_service.remove_resource('/' + srvuri) unless fetch_service.nil?
+      if fetch_service.resources.empty?
+        # if we don't call deref, we cannot start another httpserver
+        # this is a reimplementation of the cleanup_service method
+        # in Exploit::Remote::SocketServer
+        temp_service = fetch_service
+        fetch_service = nil
+        temp_service.cleanup
+        temp_service.deref
+      end
+    end
+  end
+
+  def fetch_protocol
+    'HTTPS'
+  end
+
+  def on_request_uri(cli, request, srvexe)
+    client = cli.peerhost
+    print_status("Client #{client} requested #{request.uri}")
+    if (user_agent = request.headers['User-Agent'])
+      client += " (#{user_agent})"
+    end
+    vprint_status("Client #{client} requested #{request.uri}")
+    vprint_status("Sending payload to #{client}")
+    cli.send_response(payload_response(srvexe))
```

Comment:
Should use the instance variable instead and don't pass it as an argument, for consistency?
```suggestion
    cli.send_response(payload_response(@srvexe))
```